Observer/ProductSaveAfter.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+use Klaviyo\Reclaim\Helper\Logger;
+use Klaviyo\Reclaim\Model\ProductsFactory;
+
+use Magento\Catalog\Model\CategoryFactory;
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class ProductSaveAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * Klaviyo logger helper
+     * @var \Klaviyo\Reclaim\Helper\Logger $klaviyoLogger
+     */
+    protected $_klaviyoLogger;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected $_webhookHelper;
+    protected $_categoryFactory;
+    protected $_klProductFactory;
+    protected $_stockRegistry;
+    protected $product_category_names = [];
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param ScopeSetting $klaviyoScopeSetting
+     * @param CategoryFactory $categoryFactory
+     * @param ProductsFactory $klProductFactory
+     * @param StockRegistryInterface $stockRegistry
+     * @param Logger $klaviyoLogger
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        ScopeSetting $klaviyoScopeSetting,
+        CategoryFactory $categoryFactory,
+        ProductsFactory $klProductFactory,
+        StockRegistryInterface $stockRegistry,
+        Logger $klaviyoLogger
+    ) {
+        $this->_webhookHelper = $webhookHelper;
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+        $this->_categoryFactory = $categoryFactory;
+        $this->_klProductFactory = $klProductFactory;
+        $this->_klaviyoLogger = $klaviyoLogger;
+        $this->_stockRegistry = $stockRegistry;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $product = $observer->getEvent()->getProduct();
+        $storeIds = $product->getStoreIds();
+        $storeIdKlaviyoMap = $this->_klaviyoScopeSetting->getStoreIdKlaviyoAccountSetMap($storeIds);
+
+        foreach ($storeIdKlaviyoMap as $klaviyoId => $storeIds) {
+            if (empty($storeIds)) {continue;}
+
+            if ($this->_klaviyoScopeSetting->getWebhookSecret() && $this->_klaviyoScopeSetting->getProductSaveAfterSetting($storeIds[0])) {
```

Comment:
to keep consistency with previous suggestion
```suggestion
            if ($this->_klaviyoScopeSetting->getWebhookSecret() && $this->_klaviyoScopeSetting->getProductSaveWebhookSetting($storeIds[0])) {
```